使用RedProtocol连接nb2的环境下插件发送信息时出错

报错信息：
```
11-23 12:08:31 [ERROR] nonebot_plugin_analysis_bilibili | 'MessageSegment' object has no attribute 'upload'
Traceback (most recent call last):
  File "<string>", line 33, in <module>
  File "/usr/local/lib/python3.11/site-packages/nonebot/__init__.py", line 331, in run
    get_driver().run(*args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/nonebot/drivers/none.py", line 70, in run
    loop.run_until_complete(self._serve())
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 640, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 607, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 1922, in _run_once
    handle._run()
  File "/usr/local/lib/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.11/site-packages/nonebot/message.py", line 476, in check_and_run_matcher
    await _run_matcher(
  File "/usr/local/lib/python3.11/site-packages/nonebot/message.py", line 428, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "/usr/local/lib/python3.11/site-packages/nonebot/internal/matcher/matcher.py", line 846, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "/usr/local/lib/python3.11/site-packages/nonebot/internal/matcher/matcher.py", line 821, in simple_run
    await handler(
  File "/usr/local/lib/python3.11/site-packages/nonebot/dependencies/__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "/usr/local/lib/python3.11/site-packages/nonebot_plugin_analysis_bilibili/__init__.py", line 138, in handle_analysis
    await send_msg(msg)
  File "/usr/local/lib/python3.11/site-packages/nonebot_plugin_analysis_bilibili/__init__.py", line 97, in send_msg
    await analysis_bili.send(format_msg(msg_list))
  File "/usr/local/lib/python3.11/site-packages/nonebot/internal/matcher/matcher.py", line 554, in send
    return await bot.send(event=event, message=_message, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/nonebot/adapters/red/bot.py", line 238, in send
    element_data = await message.export(self)
  File "/usr/local/lib/python3.11/site-packages/nonebot/adapters/red/message.py", line 389, in export
    resp = await seg.upload(bot)
AttributeError: 'MessageSegment' object has no attribute 'upload'
```

--- a/README.md
+++ b/README.md
@@ -31,7 +31,7 @@ _✨ NoneBot bilibili 视频、番剧解析插件 ✨_
 
 ## 使用方式
 
-私聊或群聊发送 bilibili 的小程序/链接，所有适配器均可使用，目前某些特性仅在 onebot 适配器中能够使用，例如下面可选项的几个。
+私聊或群聊发送 bilibili 的小程序/链接，所有适配器均可使用，在不支持发送图片的适配器中仅发送文字。
 
 ## 额外配置项(可选)
 
@@ -72,5 +72,3 @@ pip 安装后在 Nonebot2 入口文件（例如 bot.py ）增加：
 ```python
 nonebot.load_plugin("nonebot_plugin_analysis_bilibili")
 ```
-
-附：支持 a16 最后一版为 2.4.3

--- a/nonebot_plugin_analysis_bilibili/__init__.py
+++ b/nonebot_plugin_analysis_bilibili/__init__.py
@@ -5,7 +5,7 @@ from nonebot import on_regex, logger
 from nonebot.adapters import Event
 from nonebot.rule import Rule
 from .analysis_bilibili import config, b23_extract, bili_keyword, search_bili_by_title
-from nonebot_plugin_saa import Image, MessageFactory, MessageSegmentFactory, Text
+from nonebot_plugin_saa import MessageFactory, MessageSegmentFactory, Text, Image
 
 headers = {
     "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.69"
@@ -71,8 +71,10 @@ def flatten(container):
             yield i
 
 
-def format_msg(msg_list: List[Union[List[str], str]]):
+def format_msg(msg_list: List[Union[List[str], str]], is_plain_text: bool = False):
     flatten_msg_list = list(flatten(msg_list))
+    if is_plain_text:
+        return "".join([i for i in flatten_msg_list if not is_image(i)])
 
     msg: List[MessageSegmentFactory] = []
 
@@ -92,8 +94,9 @@ async def send_msg(msg_list: List[Union[List[str], str]]) -> None:
         return
 
     try:
-        factory = MessageFactory(format_msg(msg_list))
-        await (factory.send if interact else factory.finish)(reply=False)
+        await MessageFactory(format_msg(msg_list)).send()
+    except RuntimeError:
+        await analysis_bili.send(format_msg(msg_list, is_plain_text=True))
     except Exception as e:
         logger.exception(e)
         logger.warning(f"{msg_list}\n此次解析的内容可能被风控！")

--- a/nonebot_plugin_analysis_bilibili/__init__.py
+++ b/nonebot_plugin_analysis_bilibili/__init__.py
@@ -5,6 +5,7 @@ from nonebot import on_regex, logger
 from nonebot.adapters import Event
 from nonebot.rule import Rule
 from .analysis_bilibili import config, b23_extract, bili_keyword, search_bili_by_title
+from nonebot_plugin_saa import Image, MessageFactory, MessageSegmentFactory, Text
 
 headers = {
     "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.69"
@@ -72,20 +73,16 @@ def flatten(container):
 
 def format_msg(msg_list: List[Union[List[str], str]]):
     flatten_msg_list = list(flatten(msg_list))
-    try:
-        from nonebot.adapters.onebot.v11 import Message, MessageSegment
-    except ImportError:
-        return "".join([i for i in flatten_msg_list if not is_image(i)])
 
-    msg = Message()
+    msg: List[MessageSegmentFactory] = []
 
     for i in flatten_msg_list:
         if not i:
             continue
         elif is_image(i):
-            msg.append(MessageSegment.image(i))
+            msg.append(Image(i))
         else:
-            msg.append(MessageSegment.text(i))
+            msg.append(Text(i))
     return msg
 
 
@@ -95,7 +92,8 @@ async def send_msg(msg_list: List[Union[List[str], str]]) -> None:
         return
 
     try:
-        await analysis_bili.send(format_msg(msg_list))
+        factory = MessageFactory(format_msg(msg_list))
+        await (factory.send if interact else factory.finish)(reply=False)
     except Exception as e:
         logger.exception(e)
         logger.warning(f"{msg_list}\n此次解析的内容可能被风控！")

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "nonebot-plugin-analysis-bilibili"
-version = "2.6.3"
+version = "2.6.4"
 description = "nonebot2解析bilibili插件"
 authors = ["mengshouer"]
 license = "MIT"
@@ -13,8 +13,7 @@ keywords = ["nonebot", "nonebot2", "bilibili"]
 python = "^3.8"
 aiohttp = "^3.7"
 nonebot2 = "^2.0.0rc4"
-nonebot-adapter-onebot = { version = "^2.0.0-beta.1", optional = true }
-nonebot-plugin-send-anything-anywhere >= 0.3.2
+nonebot-plugin-send-anything-anywhere = [{ version = ">=0.3.2" }]
 
 [tool.poetry.dev-dependencies]
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,6 +14,7 @@ python = "^3.8"
 aiohttp = "^3.7"
 nonebot2 = "^2.0.0rc4"
 nonebot-adapter-onebot = { version = "^2.0.0-beta.1", optional = true }
+nonebot-plugin-send-anything-anywhere >= 0.3.2
 
 [tool.poetry.dev-dependencies]
 

